internal/listener/listener.go

```diff
@@ -220,3 +221,32 @@ func (l *Listener) DumpIncidents(w http.ResponseWriter, r *http.Request) {
 	enc.SetIndent("", "  ")
 	_ = enc.Encode(encodedIncidents)
 }
+
+func (l *Listener) DumpSchedules(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		_, _ = fmt.Fprintln(w, "GET required")
+		return
+	}
+
+	if !l.checkDebugPassword(w, r) {
+		return
+	}
+
+	l.runtimeConfig.RLock()
+	defer l.runtimeConfig.RUnlock()
+
+	for _, schedule := range l.runtimeConfig.Schedules {
+		fmt.Fprintf(w, "[id=%d] %q:\n", schedule.ID, schedule.Name)
+
+		// Iterate in 30 minute steps as this is the granularity Icinga Notifications Web allows in the configuration.
+		// Truncation to seconds happens only for a more readable output.
+		step := 30 * time.Minute
+		start := time.Now().Truncate(time.Second).Add(step)
+		for t := start; t.Before(start.Add(48 * time.Hour)); t = t.Add(step) {
```

Comment:
What do you think about making the `start` and `end` time configurable instead of using hard-coded values? Letting users optionally specify the desired start and end time would make it IMHO more flexible and one would be able to evaluate timestamps from the past as well and not just starting from now.

Apart from that, I would change the name of the handler to something like `EvaluateSchedules` since you aren't just dumping the schedules as they are, but evaluating them.

Reply:
The main intention for providing this was to allow opening this next to Icinga Notifications Web and check for consistency to simplify the review of this PR and possibly to debug inconsistencies between web and the daemon in the future. So my focus was more on keeping the implementation as simple as possible. Do you consider that functionality useful enough to justify the extra code to support it?

Reply:
Actually, I was just thinking of *was wäre, wenn :)* use cases that aren't really related to this PR, but rather for the future where you can specify an arbitrary start and end time. Though this functionality can also be implemented when it's actually needed, so you can ignore it.

Reply:
For now, I'd keep the name so that it stays consistent/grouped with the other `/dump-*` endpoints. I'm considering doing another PR that moves all of these to `/debug/*` so that their purpose is more clear and ideally handling auth centrally for all of them.